quadriga/src/main/java/edu/asu/diging/quadriga/core/data/EventGraphRepository.java

```diff
@@ -1,10 +1,15 @@
 package edu.asu.diging.quadriga.core.data;
 
+import java.util.List;
+import java.util.Optional;
+
 import org.bson.types.ObjectId;
 import org.springframework.data.mongodb.repository.MongoRepository;
 
 import edu.asu.diging.quadriga.core.model.EventGraph;
 
 public interface EventGraphRepository extends MongoRepository<EventGraph, ObjectId> {
 
+    public Optional<List<EventGraph>> findFirstByCollectionIdOrderByCreationTimeDesc(ObjectId collectionId);
```

Comment:
Note: Have modified this to findFirst since we just need the latest event graph.

Reply:
I believe with `findFirst` you can just return `Optional<EventGraph>`?